solutions/HE Twin Permutations.mdx

```diff
@@ -0,0 +1,69 @@
+---
+id: HE-twin-permutations
+source: HackerEarth
+title: Twin Permutations
+author: Andrew Wang
+---
+<Spoiler title =  "Solution">
+
+**Time Complexity:** $\mathcal{O}(N\log{N})$
+
+We can transform this problem into a graph of points. Let $x_i$ be the index of $i$ in the second array and $y_i$ be the index of $i$ in the first array for the point $(x_i, y_i)$. For each query, the answer is the length of the query minus the amount of points within the rectangle, $L_2 \le x_i \le R_2$ and $L_1 \le y_i \le R_1$. We can sweep across the $x$-axis to answer queries offline by adding the $y$-coordinates of the points we come across to a BIT.
```

Comment:
> into a graph of points.

graph is ambiguous, perhaps "set of points on the coordinate plane?"

> Let $x_i$ be the index of $i$ in the second array and $y_i$ be the index of $i$ in the first array for the point $(x_i, y_i)$.

For each $i\in [1,N]$, plot the point $(x_i,y_i)$ on the coordinate plane.